Syntax: allow ' in identifers (#737)
tweag/nickel: doc/manual/syntax.md

# Nickel Syntax

## Identifiers

> Nickel identifiers start with an alphabetic character, followed by alphanumeric characters in addition to `_` (underscore) and `'` (single quote). For example, `this-isn't-invalid` is valid syntax.

Reviewer: ```suggestion
Nickel identifiers start with an alphabetic character, followed by zero or more alphanumeric characters, `_` (underscores) or `'` (single quotes). For example, `this-isn't-invalid` is valid identifier.
```
